src/components/ChannelList/hooks/usePaginatedChannels.js

```diff
@@ -10,47 +8,51 @@ export const usePaginatedChannels = (
   options,
   activeChannelHandler,
 ) => {
-  const [channels, setChannels] = useState(Immutable([]));
+  const [channels, setChannels] = useState([]);
   const [loadingChannels, setLoadingChannels] = useState(true);
   const [refreshing, setRefreshing] = useState(true);
   const [error, setError] = useState(false);
-  const [offset, setOffset] = useState(0);
   const [hasNextPage, setHasNextPage] = useState(true);
+  let currentOffset = 0;
 
   const queryChannels = async (queryType) => {
     if (queryType === 'reload') {
+      currentOffset = 0;
```

Comment:
When filters change, we need to reset `offset` to 0, which was causing wrong query for `/channels` api!!